card/mod/pointer/lib/javascript/script_pointer_list_editor.js.coffee

```diff
@@ -40,15 +42,20 @@ $.extend decko,
     # should be (but is not) handled by slotReady
     # without this, "add another" was breaking tinymce editors in same slot
 
-    newInput.focus()
+    newInput.first().focus()
     decko.updateAddItemButton el
     decko.initPointerList newInput
 
   nextPointerInput: (lastItem)->
-    lastInput = lastItem.find 'input'
-    return lastInput if lastInput.val() == ''
+    lastInputs = lastItem.find 'input'
+    all_empty = true
+    for input in lastInputs
+        all_empty  = all_empty and $(input).val() == ''
+    return lastInputs if all_empty
+
     newItem = lastItem.clone()
     lastItem.after newItem
+    newItem.attr("data-index", parseInt(lastItem.attr("data-index") + 1))
     newItem.find 'input'
```

Comment:
We might need to get a code checker (lint?) for CoffeeScript. Some of these functions get pretty hairy.